ethcore/src/client/client.rs

```diff
@@ -1293,10 +1294,11 @@ impl snapshot::DatabaseRestore for Client {
 		let mut tracedb = self.tracedb.write();
 		self.importer.miner.clear();
 		let db = self.db.write();
-		db.restore(new_db)?;
+		db.key_value().restore(new_db)?;
```

Comment:
because blooms-db is placed in the same directory, it also moves it. Maybe we should move this function out of `KeyValueDB` interface to make it more descriptive